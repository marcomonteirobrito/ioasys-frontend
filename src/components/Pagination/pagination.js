import React from 'react';
import PropTypes from 'prop-types';

import {
  Container,
  Pages,
  Button,
  ButtonLeft,
  ButtonRight,
  PaginateContainer,
} from './styles';

/**
 * @param {string} id - Identificação unica do elemento
 * @param {string} className - ClassName do elemento
 * @param {number} page - Número da página atual
 * @param {number} pageLength - Número total de páginas
 * @param {event} onClickButtonLeft - Função chamada ao clicar no botão esquerdo
 * @param {event} onClickButtonRight - Função chamada ao clicar no botão direito
 * @param {boolean} disableButtonLeft - Desabilita o botão esquerdo
 * @param {boolean} disableButtonRight - Desabilita o botão direito
 */

const Pagination = ({
  id,
  className,
  page,
  pageLength,
  onClickButtonLeft,
  onClickButtonRight,
  disableButtonLeft,
  disableButtonRight,
}) => (
  <PaginateContainer>
    <Container id={id} className={className}>
      <Pages>
        Página <strong>{page}</strong> de <strong>{pageLength}</strong>
      </Pages>

      <Button data-testid="button-left" onClick={onClickButtonLeft}>
        <ButtonLeft disableButtonLeft={disableButtonLeft} />
      </Button>

      <Button data-testid="button-right" onClick={onClickButtonRight}>
        <ButtonRight disableButtonRight={disableButtonRight} />
      </Button>
    </Container>
  </PaginateContainer>
);

Pagination.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  page: PropTypes.number.isRequired,
  pageLength: PropTypes.number.isRequired,
  onClickButtonLeft: PropTypes.func.isRequired,
  onClickButtonRight: PropTypes.func.isRequired,
  disableButtonRight: PropTypes.bool.isRequired,
  disableButtonLeft: PropTypes.bool.isRequired,
};

Pagination.defaultProps = {
  id: 'pagination',
  className: 'pagination-container',
};

export default Pagination;
