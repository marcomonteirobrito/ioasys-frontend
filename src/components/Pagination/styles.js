import styled from 'styled-components';

import { FiChevronRight, FiChevronLeft } from 'react-icons/fi';

export const PaginateContainer = styled.div`
  display: flex;
  justify-content: flex-end;
  align-items: flex-end;
  width: 1175px;
  margin-top: 16px;

  @media (min-width: 1600px) {
    width: 1400px;
  }
`;

export const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;
export const Pages = styled.span``;

export const Button = styled.div`
  border: solid 1px rgba(51, 51, 51, 0.2);
  width: 32px;
  height: 32px;
  border-radius: 50%;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  margin: 0px 0px 0px 16px;
`;
export const ButtonLeft = styled(FiChevronLeft)`
  cursor: ${(props) => props.disableButtonLeft && 'not-allowed'};
  color: ${(props) =>
    props.disableButtonLeft ? 'rgba(51, 51, 51, 0.2)' : 'black'};
`;
export const ButtonRight = styled(FiChevronRight)`
  cursor: ${(props) => props.disableButtonRight && 'not-allowed'};
  color: ${(props) =>
    props.disableButtonRight ? 'rgba(51, 51, 51, 0.2)' : 'black'};
`;
