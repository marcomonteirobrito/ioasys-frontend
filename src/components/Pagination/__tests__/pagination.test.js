/* eslint-disable no-undef */
import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';

import Pagination from '../Pagination';

const renderPagination = () =>
  render(
    <Pagination
      page={2}
      pageLength={10}
      onClickButtonLeft={jest.fn()}
      onClickButtonRight={jest.fn()}
      disableButtonLeft={false}
      disableButtonRight={false}
    />
  );

const onClickButtonLeft = jest.fn();
const onClickButtonRight = jest.fn();
const disableButtonLeft = false;
const disableButtonRight = false;

describe('Pagination', () => {
  it('Deve renderizar o componente corretamente', () => {
    renderPagination();

    expect(document.body).toMatchSnapshot();
  });

  it('Deve chamar onClick ao clicar no botão esquerdo', () => {
    render(
      <Pagination
        page={1}
        pageLength={10}
        onClickButtonLeft={onClickButtonLeft}
        onClickButtonRight={onClickButtonRight}
        disableButtonLeft={disableButtonLeft}
        disableButtonRight={disableButtonRight}
      />
    );

    const buttonLeft = screen.getByTestId('button-left');
    fireEvent.click(buttonLeft);

    expect(onClickButtonLeft).toHaveBeenCalledTimes(1);
    expect(onClickButtonRight).toHaveBeenCalledTimes(0);
  });

  it('Deve chamar onClick ao clicar no botão direito', () => {
    render(
      <Pagination
        page={1}
        pageLength={10}
        onClickButtonLeft={onClickButtonLeft}
        onClickButtonRight={onClickButtonRight}
        disableButtonLeft={disableButtonLeft}
        disableButtonRight={disableButtonRight}
      />
    );
    const buttonRight = screen.getByTestId('button-right');
    fireEvent.click(buttonRight);

    expect(onClickButtonRight).toHaveBeenCalledTimes(1);
    expect(onClickButtonLeft).toHaveBeenCalledTimes(0);
  });
});
