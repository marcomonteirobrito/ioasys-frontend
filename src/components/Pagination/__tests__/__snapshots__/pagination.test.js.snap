// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Deve renderizar o componente corretamente 1`] = `
<body>
  <div>
    <div
      class="sc-bdfBwQ enbGoq pagination-container"
      id="pagination"
    >
      <span
        class="sc-gsTCUz"
      >
        PÃ¡gina 
        <strong>
          2
        </strong>
         de 
        <strong>
          10
        </strong>
      </span>
      <div
        class="sc-dlfnbm dvbuTo"
        data-testid="button-left"
      >
        <svg
          class="sc-hKgILt jCnWdz"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="15 18 9 12 15 6"
          />
        </svg>
      </div>
      <div
        class="sc-dlfnbm dvbuTo"
        data-testid="button-right"
      >
        <svg
          class="sc-eCssSg kmNnxw"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="9 18 15 12 9 6"
          />
        </svg>
      </div>
    </div>
  </div>
</body>
`;
